### ISSUE TITLE
sets with multiple midi inputs do not save/reload connections properly

### ISSUE THREAD

**Issue by twhiston** [Cycling74/rnbo-runner-panel#180]
I have a setup where I have 3 midi controllers connected to my rpi. Two of these are Novation Launch Controls, and additionally a keystation mini. I'm using RAW midi mapping and these show up as

in-hw-5-0-0-Keystation-Mini-32
in-hw-6-0-0-Launch-Control-XL
in-hw-6-0-1-Launch-Control-XL
in-hw-8-0-0-Launch-Control-XL
in-hw-8-0-1-Launch-Control-XL

of these I use in-hw-5-0-0-Keystation-Mini-32, in-hw-6-0-0-Launch-Control-XL and in-hw-8-0-0-Launch-Control-XL

When I save and reload my set and preset it retains the connection from in-hw-8-0-0-Launch-Control-XL to [rnbo-control] but everything else has moved to be connected to in-hw-6-0-0-Launch-Control-XL!

Is there something I can send you to help diagnose why this is the case?
I guess it's because they are the same type of midi controller?

**Comment by x37v** [comment]
yeah, unfortunately I'm not sure how to make this work without unique identifiers for each device

**Comment by x37v** [comment]
i guess at least if there are multiple devices with the same name we could connect to all of them instead of just 1

**Comment by twhiston** [comment]
That would definitely allow me to work around my current issues with some clever midi mapping on each controller. At the moment I've had to put an iconnectivity mioXM in between so that I can use its fixed output ports to ensure everything is connected properly, but it's a bit of unnecessary gear to travel with really and i'd love to drop it.
